packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/JavaDriverDialectParser.kt

```diff
@@ -110,6 +90,40 @@ object JavaDriverDialectParser : DialectParser<PsiElement> {
             emptyList()
         }
 
+    override fun isReferenceToDatabase(source: PsiElement): Boolean {
+        val refToDb =
+            source
+                .parentOfType<PsiMethodCallExpression>(true)
+                ?.findMongoDbClassReference(source.project)
+                ?: return false
+
+        return refToDb.type?.isMongoDbDatabaseClass(refToDb.project) == true
+    }
+
+    override fun isReferenceToCollection(source: PsiElement): Boolean {
+        val refToDb =
+            source
+                .parentOfType<PsiMethodCallExpression>(true)
+                ?.findMongoDbClassReference(source.project)
+                ?: return false
+
+        return refToDb.type?.isMongoDbCollectionClass(refToDb.project) == true
+    }
+
+    override fun isReferenceToField(source: PsiElement): Boolean {
+        val isInQuery = isInQuery(source)
```

Comment:
Interesting that this just works despite declaring the identifier the same as the function name.

Reply:
Yeah, it's cool when the language has separate namespaces for functions and variables.

Reply:
ohh so that's how .. things keep getting interesting about kotlin :D